### ISSUE TITLE
series request limit can be bypassed by users

### ISSUE THREAD

**Issue by solidsnakex37** [seerr-team/seerr#633]
### Description

If you have a "Series Request Limit" set for a user, for example

4 Seasons per 7 days 

The user is able to bypass the season limit when you have the "Auto Approve" setting disabled. 

While it's in the "Pending State" awaiting admin approval, the user can go back in and edit the request and select more seasons than they are allowed. 

### Version

1.7.0

### Steps to Reproduce

1. Create a test user 
2. Assign a limit to this user for seasons. For example, allow 4 season requests per 7 days. Note: Assign a limit to the user using the Override Global Limit option, don't assign a limit to the user just using the global limit. (General > Users > Select User > General Tab > Series Request Limit > Override Global Limit)
3. Disable "Auto Approve" for this user, so that it requires admin approval. 
4. On the test user, request a show that has more than 4 seasons, but select all that you're allotted to (4 seasons), Then submit
5. Once submitted, head to the requests tab, edit your request, then select the remaining seasons and click save. 

### Screenshots

![image](https://github.com/Fallenbagel/jellyseerr/assets/109041152/b51ab021-8cec-4f99-b3cd-5550e3906b52)
![image](https://github.com/Fallenbagel/jellyseerr/assets/109041152/e8f681bd-7951-4bda-83c0-a54c8b7f2c63)
![image](https://github.com/Fallenbagel/jellyseerr/assets/109041152/2a53772a-eae5-4baf-abbe-2a212e48682e)
![image](https://github.com/Fallenbagel/jellyseerr/assets/109041152/35eb8502-6dcf-43b2-82b5-e58ff807b071)
![image](https://github.com/Fallenbagel/jellyseerr/assets/109041152/a82b4ad5-d3f6-4312-8840-3ea8b1f4f937)

### Platform

desktop

### Device

Any

### Operating System

Unraid

### Browser

Any

### Code of Conduct

- [X] I agree to follow Jellyseerr's Code of Conduct

**Comment by Fallenbagel** [comment]
Can you try on develop

**Comment by solidsnakex37** [comment]
Same issue

**Comment by solidsnakex37** [comment]
![image](https://github.com/Fallenbagel/jellyseerr/assets/109041152/d1c69b68-27e6-4ff2-885b-60f45b7119fa)

**Comment by Danish-H** [comment]
I am unable to recreate this issue on develop. When I try to edit the request, it expectedly still enforces the limit. Am I missing something?

![image](https://github.com/Fallenbagel/jellyseerr/assets/121830048/94e5a02c-4a28-4a61-806b-e68a168db53e)

**Comment by solidsnakex37** [comment]
I just updated to the latest develop and still see the issue, I don't have the "No season requests remaining" prompt when I edit a request, I am not sure how you have that

Are you overriding the global limit for your user? (Step 2 in the replication steps). I will update this in case there was confusion
![image](https://github.com/Fallenbagel/jellyseerr/assets/109041152/352dc743-f1a5-4bed-a157-ef0892ecf1f1)



![image](https://github.com/Fallenbagel/jellyseerr/assets/109041152/3571ef01-2218-4778-bb3b-70e535d0be59)

**Comment by solidsnakex37** [comment]
> I am unable to recreate this issue on develop. When I try to edit the request, it expectedly still enforces the limit. Am I missing something?
> 
> ![image](https://private-user-images.githubusercontent.com/121830048/300599438-94e5a02c-4a28-4a61-806b-e68a168db53e.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MDY1NjQ2MjMsIm5iZiI6MTcwNjU2NDMyMywicGF0aCI6Ii8xMjE4MzAwNDgvMzAwNTk5NDM4LTk0ZTVhMDJjLTRhMjgtNGE2MS04MDZiLWU2OGExNjhkYjUzZS5wbmc_WC1BbXotQWxnb3JpdGhtPUFXUzQtSE1BQy1TSEEyNTYmWC1BbXotQ3JlZGVudGlhbD1BS0lBVkNPRFlMU0E1M1BRSzRaQSUyRjIwMjQwMTI5JTJGdXMtZWFzdC0xJTJGczMlMkZhd3M0X3JlcXVlc3QmWC1BbXotRGF0ZT0yMDI0MDEyOVQyMTM4NDNaJlgtQW16LUV4cGlyZXM9MzAwJlgtQW16LVNpZ25hdHVyZT01YzZiYWExZjA5NWJjYjQ4MzVmZTBhZjQ0MGFiNWM5N2I3ODE1ZGJmYTM3NmFkY2EzNzg0MmQ3MzczNzUyZmExJlgtQW16LVNpZ25lZEhlYWRlcnM9aG9zdCZhY3Rvcl9pZD0wJmtleV9pZD0wJnJlcG9faWQ9MCJ9.C3BJbvvmNsRwsL7kMbzsKNRufE_bbkdUzAQwtppiTR8)

I've found the root setting that's causing this issue,  you have to have "Advanced Requests" enabled for the user as well. When I unchecked this permission from my test user, I see the series limit just as you would expect, per your screenshot. Once I check that option, I can bypass my series limit by editing the request. 

![image](https://github.com/Fallenbagel/jellyseerr/assets/109041152/e06884c6-edeb-480b-9546-fef681162257)

**Comment by Danish-H** [comment]
Thank you for sharing that, it really helped

The issue seems to be originating from here. The value of `quota` is initially correct, but it immediately changes to null because of the `requestOverrides?.user?.id` condition becoming true

https://github.com/Fallenbagel/jellyseerr/blob/db84f6529ab285be26c96daaab065dfabf347417/src/components/RequestModal/TvRequestModal.tsx#L90-L95

Removing that entire condition worked for me, but I wanted to ask, is it there for a reason? Should we not be enforcing quota on every user regardless of these conditions?

**Comment by Fallenbagel** [comment]
> Removing that entire condition worked for me, but I wanted to ask, is it there for a reason? Should we not be enforcing quota on every user regardless of these conditions?

Good question. I'll ask overseerr about this. I feel like advanced request is still not a user level that should get the quotas to be null

**Comment by Fallenbagel** [comment]
> > > Removing that entire condition worked for me, but I wanted to ask, is it there for a reason? Should we not be enforcing quota on every user regardless of these conditions?
> > 
> > Good question. I'll ask overseerr about this. I feel like advanced request is still not a user level that should get the quotas to be null
> 
> Did they ever get back to you on this? 
> 
> 

Yes. Not intended behavior. I have a fix, I'll be opening a pr on overseerr

**Comment by ghost** [comment]
I noticed that there wasn't an active PR addressing this issue yet on Overseerr or Seerr, even though you've essentially already described the solution here. Since I'm new to the OpenSource and would like to contribute, I created [the missing PR](https://github.com/seerr-team/seerr/pull/2190). Should I also create one on Overseerr? If I've overlooked something or if anything else is incorrect, please let me know.

**Comment by fallenbagel** [comment]
>  Should I also create one on Overseerr? If I've overlooked something or if anything else is incorrect, please let me know.

No. Overseerr repo is deprecated. We are merged now.

**Comment by jboss38** [comment]
Is this issue has been "fixed" with last version 3.0.1 ?

**Comment by fallenbagel** [comment]
> Is this issue has been "fixed" with last version 3.0.1 ?

https://github.com/seerr-team/seerr/pull/2190

**Comment by Logvin** [comment]
I'd like to take this one on. The root cause is clear from the discussion above - when AdvancedRequester sets requestOverrides.user, the SWR quota key becomes null for users who have REQUEST_ADVANCED but not MANAGE_USERS, so quota is never fetched or enforced.

I'm planning to split this into two PRs:

PR 1 (frontend): Fix the SWR condition in all three request modals (TV, Movie, Collection) so quota is always fetched unless the user has MANAGE_REQUESTS. This is the same direction Danish-H identified above. Small diff, 3 files.

PR 2 (backend): Add server-side quota enforcement on the PUT /request/:id endpoint so it can't be bypassed even with a modified client. This one needs more extensive testing so I want to get the frontend fix checked off first.

I saw PR #2190 went stale with a similar approach. Happy to adjust if there's a preferred direction.

**Comment by Sir-Doyle** [comment]
Hi

This issue still there in the version 3.4.1

**Comment by fallenbagel** [comment]
> Hi
> 
> This issue still there in the version 3.4.1 

If this issue was solved this will be closed
